src/compiler/crystal/command/env.cr

```diff
@@ -7,9 +7,10 @@ class Crystal::Command
     end
 
     vars = {
-      "CRYSTAL_CACHE_DIR" => CacheDir.instance.dir,
-      "CRYSTAL_PATH"      => CrystalPath.default_path,
-      "CRYSTAL_VERSION"   => Config.version || "",
+      "CRYSTAL_CACHE_DIR"    => CacheDir.instance.dir,
+      "CRYSTAL_PATH"         => CrystalPath.default_path,
+      "CRYSTAL_VERSION"      => Config.version || "",
```

Comment:
Could this ever be nil?

Reply:
Crystal.version probably not since now there is a file with it. This line is from time where git describe was used. It can be handled in another PR if wanted

Reply:
I agree. I think that's better than just `CRYSTAL_VERSION=` and Crystal developers will probably understand `CRYSTAL_VERSION=nil`.
```suggestion
      "CRYSTAL_VERSION"      => Config.version || "nil",
```

Reply:
`nil` isn't a proper semantic version...

Reply:
It should just indicate that there is no semantic version.

Reply:
Or leave it as `""` with a BIG `TODO: Remove nil condition` ? :stuck_out_tongue:

Reply:
Actually the version can be nil, see #7602

Reply:
I didn't expect that. seems like config should not allow `nil` to be returned.
https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/config.cr#L7

Also. This might want to be its own PR.